File: algorithms/3838.swift
// 3838. Weighted Word Mapping
// https://leetcode.com/problems/weighted-word-mapping/description/?envType=daily-question&envId=2026-06-13

class Solution {
    func mapWordWeights(_ words: [String], _ weights: [Int]) -> String {
        var answer = ""

        for word in words {
            var sum = 0

            for ch in word {
                let idx = Int(ch.asciiValue!) - Int(Character("a").asciiValue!)
                sum += weights[idx]
            }

            sum %= 26

            let reversedChar = Character(
                UnicodeScalar(
                    Int(Character("a").asciiValue!) + (25 - sum)
                )!
            )

            answer.append(reversedChar)
        }

        return answer
    }
}

File: algorithms/3838.py
# 3838. Weighted Word Mapping
# https://leetcode.com/problems/weighted-word-mapping/?envType=daily-question&envId=2026-06-13

class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        answer = ""
        for word in words:
            result = 0
            for ch in word:
                idx = ord(ch) - ord("a")
                result += weights[idx]
            result = result % 26
            answer_idx = ord("a") + (25 - result)
            answer += chr(answer_idx)

        return answer
